stored_requests/config/config.go

```diff
@@ -42,32 +42,30 @@ var dbConnection struct {
 //
 // As a side-effect, it will add some endpoints to the router if the config calls for it.
 // In the future we should look for ways to simplify this so that it's not doing two things.
-func CreateStoredRequests(cfg *config.StoredRequestsSlim, metricsEngine pbsmetrics.MetricsEngine, client *http.Client, router *httprouter.Router) (fetcher stored_requests.AllFetcher, db *sql.DB, shutdown func()) {
+func CreateStoredRequests(cfg *config.StoredRequestsSlim, metricsEngine pbsmetrics.MetricsEngine, client *http.Client, router *httprouter.Router, dbc *dbConnection) (fetcher stored_requests.AllFetcher, shutdown func()) {
```

Comment:
The change of no longer returning the DB connection makes the comment above the function incorrect (line 37).  This gives me the opportunity to propose that maybe the function should retain the old behavior of returning the DB connection rather than modifying the structure pointed to in the new argument (which would then not need to be a pointer).  I find it a little confusing that three things are set or changed by this function but only two of the changes are available via return value and the other is in one of the function arguments.  Either way a change is need to update either the function description or the function itself.

Reply:
I will have to fix the comment. We need the `dbConnection` object so we can confirm that we can reuse the same db object (verify the connection string). It is also used in the shutdown functions in order to avoid trying to shut down the same database multiple times.

Basically `dbConnection` was a global mutable variable, which could have caused issues in testing. It is only used by `CreateStoredRequests()` and `NewStoredRequests()`. So I made it local to `NewStoredRequests()` and passed it on to `CreateStoredRequests()` to avoid issues with mutable global variables.

Reply:
My two cents on this:

@hhhjort @josephveach What do you think about breaking this function up into two functions: `NewDBConn` and `CreateFetcher` or something along those lines? I think the confusion is arising from the fact that this function is doing both of those things and IMHO, the function name `CreateStoredRequests` doesn't convey either of those. I am not sure if this would be as simple as splitting the method in to two and then in the `NewStoredRequests` function calling the `NewDBConn` function followed by `CreateFetchers` for various fetchers or a little more complicated than that based on where all is this function used. But nevertheless, if it's the former than it might be worth doing.

Thinking of the `dbConn` as a singleton, I think that approach would make more sense and then it becomes clearer to pass the `dbConn` object to `CreateFetcher` and why it'd need it. Also, at that point `CreateFetcher` won't modify the `dbConn` object at all. It will just use it. 

And lastly, of course if it is too much effort to do in this PR then we can always do it as a separate one. But, would love to know what do you guys think about this?

Reply:
I agree that this code has been increasing in complexity and could be changed to make it more readable and intuitive.  It may be better to wait for it to reach equilibrium rather than pouring a lot of effort into it now and risking needing to redesign it again for the next update.

Reply:
I would say refactoring it out into two functions might be best for a new PR. DOn't want the scope of this to creep too far, especially after having it sit pending for so long.

Reply:
Sounds reasonable!